src/west/manifest.py

```diff
@@ -832,8 +832,8 @@ def sha(self, rev: str, cwd: Optional[PathType] = None) -> str:
         # Though we capture stderr, it will be available as the stderr
         # attribute in the CalledProcessError raised by git() in
         # Python 3.5 and above if this call fails.
-        cp = self.git(f'rev-parse {rev}', capture_stdout=True, cwd=cwd,
-                      capture_stderr=True)
+        cp = self.git(f'rev-parse {rev}^{{commit}}', capture_stdout=True,
```

Comment:
Why is that?

```
carles@ubuntu ~/src/zephyr/zephyr (main)
$ git rev-parse v2.6.0
d477a71920b3fd4491dbef3f29906f2421ad2329
```

Reply:
> git rev-parse v2.6.0
> d477a71920b3fd4491dbef3f29906f2421ad2329

whereas:
```
git rev-parse v2.6.0^{commit}
79a6c07536bc14583198f8e3555df6134d8822cf
```

Reply:
Ah right. This is only for annotated tags then. Thanks!

Reply:
> $ git rev-parse v2.6.0
> d477a71920b3fd4491dbef3f29906f2421ad2329

To expand on what @marc-hb said,

```
$ git cat-file -t d477a71920b3fd4491dbef3f29906f2421ad2329
tag
$ git cat-file -t 79a6c07536bc14583198f8e3555df6134d8822cf
commit
```

d4777a7 is the SHA of the tag object, not the commit object.